### PR TITLE
feat: add `useDocumentNavigation` hook

### PR DESCRIPTION
**Pull request by oliverschwendener**
This PR adds the react hook for getting the document navigation items.

### CODE REVIEW

File: packages/app-bridge/src/tests/DocumentNavigationTreeDummy.ts [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+    type: 'page-category';
+
+    private constructor(
+        private readonly _id: number,
+        private readonly _title: string,
+        private readonly _slug: string,
+        private readonly _children: (GuidelineDocumentPage | GuidelineDocumentPageLink)[],
+    ) {
+        this.type = 'page-category';
+    }
```
Comment (anxobotana) [inline packages/app-bridge/src/tests/DocumentNavigationTreeDummy.ts:35]: Let's define these outside the constructor:

```suggestion
    readonly type = 'page-category' as const;
    readonly #id: number;
    readonly #title: string;
    readonly #slug: string;
    readonly #children: (GuidelineDocumentPage | GuidelineDocumentPageLink)[];
    
   constructor(id: number, title: string, slug: string, children: (GuidelineDocumentPage | GuidelineDocumentPageLink)[],
    ) {
        this.#id = id;
        this.#title = title;
        this.#slug = slug;
        this.#children = children;
    }
```

same thing for the others

File: packages/app-bridge/src/tests/DocumentNavigationTreeDummy.ts [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+
+    id(): number {
+        return this._id;
```
Comment (anxobotana) [inline packages/app-bridge/src/tests/DocumentNavigationTreeDummy.ts:38]: then you need to use:

```suggestion
        return this.#id;
```

**Review by anxobotana — CHANGES_REQUESTED** [review]
just the changes in the dummy item

File: packages/app-bridge/src/tests/DocumentNavigationTreeDummy.ts [hunk 3]
```diff
@@ -9,195 +9,136 @@ import {
...
-    }: {
-        id?: number;
-        title?: string;
-        slug?: string;
-        children?: (GuidelineDocumentPage | GuidelineDocumentPageLink)[];
-    }): GuidelinePageCategory {
-        return new DocumentPageCategoryDummy(id ?? 1234, title ?? 'Dummy Title', slug ?? 'dummy-slug', children ?? []);
-    }
-
     type: 'page-category';
```
Comment (anxobotana) [inline packages/app-bridge/src/tests/DocumentNavigationTreeDummy.ts:12]: if you define this here like this, you don't need to set the value in the constructor, it looks redundant:
```
readonly type = 'page-category' as const;
```

Comment (oliverschwendener) [reply packages/app-bridge/src/tests/DocumentNavigationTreeDummy.ts]: done

**Review by anxobotana — APPROVED** [review]
LGTM 💪

**Review by olivereisenhut — APPROVED** [review]

**Pull request merged**